### PR TITLE
4 goals are supported now in algorithm and maze class.

### PR DESCRIPTION
**Pull request by gonzagab**
Made it so that the algorithm can support 4 goals. Need to test that when reading a maze from a file that the goals are set properly. Also destinations still need to be added to the MouseController.

### CODE REVIEW

**Comment by gonzagab** [comment]
#32 is the issue being solved here.

File: source/maze/MazeIO.cpp [hunk 1]
```diff
@@ -89,9 +89,9 @@ void MazeIO::loadMaze(char* fileName)
                 maze->startPos.col = j / 2;
                 wallToPut = wallError;
                 continue;
-            case 'G':
-                maze->goalPos.row = i / 2;
-                maze->goalPos.col = j / 2;
+            case 'G':		/* needs to be tested */
```
Comment (kbumsik) [inline source/maze/MazeIO.cpp]: This is *definitely* no-no! Don't try ~~lazy~~ workaround.
You must not assume there are 4 goal in a given maze file.
when reading a maze, you must empty `goalPos` and fill fill goals by using [`vector::push_back()`](http://www.cplusplus.com/reference/vector/vector/push_back/)

File: source/maze/MazeIO.cpp [hunk 2]
```diff
@@ -89,9 +89,9 @@ void MazeIO::loadMaze(char* fileName)
...
+                maze->goalPos[(i%5) + (j%2)].row = i / 2;
+                maze->goalPos[(i%5) + (j%2)].col = j / 2;
```
Comment (kbumsik) [inline source/maze/MazeIO.cpp]: This is *definitely* no-no! Don't try easy ~~lazy~~ workaround.
You must not assume there are 4 goal in a given maze file.
when reading a maze, you must empty `goalPos` and fill fill goals by using [`vector::push_back()`](http://www.cplusplus.com/reference/vector/vector/push_back/)

**Comment by gonzagab** [comment]
Ok sounds good. Ill get on it.

**Pull request merged**